### DISCUSSION TITLE
HELP！I can't get responsed while using homekit adk

### DISCUSSION THREAD

**Discussion by moculll** [openthread/openthread#9439] [Q&A] [answered]
![image](https://github.com/openthread/openthread/assets/101570525/668d9d1a-77cd-4a75-b9ab-a066e6656706)
As you can see above, this is the normal case, It works ,and the apple home can control the device well
and when I add a few more services, the ot bufferinfo shows like this:
![image](https://github.com/openthread/openthread/assets/101570525/9f7f585a-289f-4769-ba23-6862b311dd69)
maybe is the coap overflow or something.....? I don't know, I've tried adding `CONFIG_OPENTHREAD_CUSTOM_PARAMETERS=" OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE=1 OPENTHREAD_CONFIG_MESSAGE_BUFFER_SIZE=138"` to my project, but nothing changed, what I can make sure is that adding services will cause this problem, please help me T T

**Comment by jwhui** [comment, marked as answer]
It appears that the CoAP component is holding on to messages. Note that CoAP normally buffers confirmable messages to properly match responses. It could be that the device is not receiving CoAP responses, or at least not at the rate that it is trying to send requests.

Unfortunately, I don't have visibility into how the HomeKit ADK leverages CoAP. I would reach out to Apple or whoever provided you the HomeKit ADK for more information.

**Reply by moculll** [reply]
@jwhui Thank you, not the same problem have seen in matter thread network, I guess that is the apple's problem
